tests/integration/tier0/non-destructive/file-backup/test_file_backup.py

```diff
@@ -3,90 +3,151 @@
+        modified_file_path = os.path.join(MODIFIED_FILES_DIR, f"{file_name}.modified")
+        bkp_file_path = os.path.join(BACKUP_DIR, file_name)
         # Install the packages if not installed already
         if "is not installed" in shell(f"rpm -q {pkg}").output:
             shell(f"yum install -y {pkg}")
-        # If we check for file modification being in place after the rollback
-        if os.environ.get("C2R_TESTS_MODIFIED_CONFIGS"):
-            # Append the modified content to the config
-            # The file will be created if not already
-            with open(config, "a+") as cfg:
-                cfg.write(modified_content)
-                cfg.seek(0)
-                modified_file_data = cfg.read()
-        else:
-            # Remove the config to validate it won't get restored
-            # during the rollback
-            shell(f"rm -f {config}")
+
+        # Create the config file, if not present already
+        if not os.path.exists(config):
+            shell(f"touch {config}")
+        # Backup the original file
+        shell(f"cp {config} {BACKUP_DIR}")
+
+        # Append the modified content to the config
+        # The file will be created if not already
+        with open(config, "a+") as cfg:
+            cfg.write(MODIFIED_CONTENT)
+            cfg.seek(0)
+            # modified_file_data = cfg.read()
+        # Copy the modified file for later comparison
+        shell(f"cp {config} {modified_file_path}")
+
+        backup_paths[file_name] = [modified_file_path, bkp_file_path, config]
 
     yield
 
-    for pkg, config in packages.items():
-        # Verify the packages are still installed
-        assert shell(f"rpm -q {pkg}").returncode == 0
-        # Verify the pre-conversion and post-conversion
-        if os.environ.get("C2R_TESTS_MODIFIED_CONFIGS"):
-            # Verify the config file got restored during rollback
-            assert os.path.exists(config)
-            # Read the restored content
-            with open(config, "r") as cfg:
-                restored_file_data = cfg.read()
-
-            # Verify the content is same
-            assert modified_file_data == restored_file_data
-        else:
-            # Verify the config did not get restored if not present
-            # prior the conversion
-            assert not os.path.exists(config)
+    for file, paths in backup_paths.items():
+        modified_file_path = paths[0]
+        bkp_file_path = paths[1]
+        default_config_path = paths[2]
+        # Verify the config file got restored during rollback
+        assert os.path.exists(default_config_path)
+        # Read the restored content
+        with open(default_config_path, "r") as cfg:
+            restored_file_data = cfg.read()
+        with open(modified_file_path, "r") as orig_cfg:
+            modified_file_data = orig_cfg.read()
+
+        # Verify the content is same
+        assert modified_file_data == restored_file_data
```

Comment:
Maybe we can use some function. Either python function `filecmp.cmp()` (https://docs.python.org/3/library/filecmp.html) or use just the `shell(diff file1 file2)`. It can save few lines of code

Reply:
+1 to filecmp.cmp()... Just learned about the existence of this right now.